### ISSUE TITLE
offsetTrail' shift the origin in diagrams-2.0

### ISSUE THREAD

**Issue by wkoiking** [cchalmers/diagrams-dev#7]
Following code in diagrams-2.0 produces the below image:
~~~
renderTest :: IO ()
renderTest = saveDiagram Cairo "CCC.png" (mkWidth 1000) $ showOrigin $ showEnvelope $ tri
 where tri = strokeLocTrail $ offsetTrail' (with & offsetJoin .~ LineJoinRound) 0.3
                            $ (triangle 2 :: Located (Trail V2 Double))

strokeLocTrail :: Located (Trail V2 Double) -> Diagram V2
strokeLocTrail = strokePath . fromLocTrail
~~~
![CCC](https://user-images.githubusercontent.com/3189740/56136521-962e2480-5fb0-11e9-9c86-fb75d419de92.png)

Whereas the similar code below in diagrams-1.4 produces the following image:
~~~
renderTest :: IO ()
renderTest = renderCairo "YYY.png" (mkWidth 1000) $ showOrigin $ showEnvelope $ tri
 where tri = strokeLocTrail $ offsetTrail' (with & offsetJoin .~ LineJoinRound) 0.3
                            $ (triangle 2 :: Located (Trail V2 Double))
~~~
![YYY](https://user-images.githubusercontent.com/3189740/56136499-8adaf900-5fb0-11e9-9fcc-82c82cbe069e.png)

**Comment by byorgey** [comment]
The only thing I can see in `offsetTrail'` that changed between `diagrams-lib` and `geometry` is that `fromLocTrail` is used in place of `trailPoints`.  Compare:

https://github.com/diagrams/geometry/blob/master/src/Geometry/TwoD/Offset.hs#L323

https://github.com/diagrams/diagrams-lib/blob/master/src/Diagrams/TwoD/Offset.hs#L287

However, it should be using the `[Point v n]` instance of `FromTrail`, which is defined to call `trailPoints`: https://github.com/diagrams/geometry/blob/master/src/Geometry/Trail.hs#L960 .  So I don't yet know the reason for the difference.

Are you sure the difference is introduced by `offsetTrail'`?  i.e. do you get the same trails before you call that function?

**Comment by wkoiking** [comment]
> Are you sure the difference is introduced by offsetTrail'? i.e. do you get the same trails before you call that function?

`strokeLocTrail (triangle 2 :: Located (Trail V2 Double))` produces
![CCC](https://user-images.githubusercontent.com/3189740/56803875-d453ff80-685e-11e9-933c-7e81e126b655.png)
in both diagrams-2.0 and diagrams-1.4, so I think only possible cause is `offsetTrail'`.

**Comment by cchalmers** [comment]
Sorry it took me so long to get to this, I was busy over easter. This bug was caused by `dirBetween` changing (diagrams/geometry@007c890) but the places where it got used weren't updated. I think it should be fixed now (diagrams/geometry@490bf4c).

**Comment by cchalmers** [comment]
I've added a couple of small offset tests to diagrams/diagrams-backend-tests@8adf45c. We should really have more tests for things like this so issues get caught earlier.

![image](https://user-images.githubusercontent.com/1354887/56866859-5450a580-69d6-11e9-8da0-0127953cbfc7.png)
